api/v1/controllers/collectors.js

```diff
@@ -77,6 +79,59 @@ function reEncryptSGContextValues(sg, authToken, timestamp) {
 }
 
 /**
+ * Find the matching generator template and attach it to the generator
+ *
+ * @param  {Object} sg - Sample generator object
+ * @return {Object} Sample generator with generatorTemplate attribute set to the
+ *  full matching generator template object.
+ */
+function attachTemplate(sg) {
+  const { name, version } = sg.generatorTemplate;
+  return GeneratorTemplate.getSemverMatch(name, version)
+  .then((gt) => {
+    if (sg.context && gt.contextDefinition) {
+      sg = JSON.parse(JSON.stringify(sg));
```

Comment:
Any reason why a new copy of "sg" is being created?

Reply:
It's possible for the same generator object to be present in multiple places in the collectorMap. This is necessary to prevent it from being encrypted twice.

When I originally wrote this things were structured differently and this was the best place to put it, but now it definitely makes more sense to do this in trackGeneratorChanges before the generator object is added to the collectorMap. Thanks for catching this, I will move it.